mobile/test/common/integration/base_client_integration_test.h

```diff
@@ -61,17 +62,24 @@ class BaseClientIntegrationTest : public BaseIntegrationTest {
   envoyToMobileHeaders(const Http::TestRequestHeaderMapImpl& request_headers);
   Event::ProvisionalDispatcherPtr dispatcher_ = std::make_unique<Event::ProvisionalDispatcher>();
   envoy_http_callbacks bridge_callbacks_;
-  ConditionalInitializer terminal_callback_;
-  callbacks_called cc_{0, 0, 0, 0, 0, 0, 0, "", &terminal_callback_, {}};
+  std::shared_ptr<ConditionalInitializer> terminal_callback_;
```

Comment:
this doesn't need to be a shared_ptr, it can be a reference, i.e. `ConditionalInitializer&`.  You can assign `terminal_callback = multi_terminal_callbacks[0];` and it'll store the first entry in the array by reference to `ConditionalInitializer& terminal_callback`.

Reply:
I tried that but got the following error, I think it's because references cannot be initialized as null? ```test/common/integration/base_client_integration_test.cc:86:28: error: constructor for 'Envoy::BaseClientIntegrationTest' must explicitly initialize the reference member 'terminal_callback_'
BaseClientIntegrationTest::BaseClientIntegrationTest(Network::Address::IpVersion ip_version,
                           ^
./test/common/integration/base_client_integration_test.h:60:27: note: declared here
  ConditionalInitializer& terminal_callback_;
                          ^
test/common/integration/base_client_integration_test.cc:86:28: error: constructor for 'Envoy::BaseClientIntegrationTest' must explicitly initialize the reference member 'cc_'
BaseClientIntegrationTest::BaseClientIntegrationTest(Network::Address::IpVersion ip_version,
```

Reply:
We settled on using unique and raw pointers